Runtime/Scripts/Texture2DDisposable.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Unity.MeshSync {
+    /// <summary>
+    /// Wrapper class for Texture2D to handle cleaning up resources when the texture is
+    /// no longer required.
+    /// </summary>
+    internal class Texture2DDisposable : IDisposable {
+        public Texture2D Texture { get; private set; }
+
+        private bool DoDispose { get; }
```

Comment:
is this actually used anywhere or is it dead code?

Reply:
It's used in Dispose() to check whether the texture should be disposed or not.